File: src/HookBridge.Application/ControlPlane/UseCases/Documentation/DocModels.cs
namespace HookBridge.Application.ControlPlane.UseCases.Documentation;

public sealed record DocSectionDto(
    string Id,
    string Title,
    string Category,
    string Summary,
    string ContentMarkdown,
    Dictionary<string, string> CodeSnippets
);

public sealed record DocEndpointDto(
    string Id,
    string Category,
    string Method,
    string Path,
    string Summary,
    string Description,
    List<DocParamDto> PathParameters,
    List<DocParamDto> QueryParameters,
    List<DocHeaderDto> RequestHeaders,
    string? SampleRequestBody,
    int ExpectedStatusCode,
    string? SampleResponseBody,
    Dictionary<string, string> CodeSnippets
);

public sealed record DocParamDto(
    string Name,
    string Type,
    bool Required,
    string Description,
    string? Example
);

public sealed record DocHeaderDto(
    string Name,
    bool Required,
    string Description,
    string Example
);

public sealed record SdkRecipeDto(
    string Language,
    string DisplayName,
    string Description,
    string InstallationCommand,
    string VerificationSnippet,
    string PublishingSnippet,
    List<string> Dependencies
);

public sealed record ApiReferenceResponse(
    string Version,
    string Environment,
    string BaseUrl,
    List<DocSectionDto> Guides,
    List<DocEndpointDto> Endpoints,
    List<SdkRecipeDto> SdkRecipes
);

public sealed record GenerateDocSnippetRequest(
    string Language,
    string Method,
    string Path,
    Dictionary<string, string>? Headers,
    string? Body,
    string? ApiKey,
    string? SigningSecret
);

public sealed record GenerateDocSnippetResponse(
    string Language,
    string Snippet,
    string FormattedCurl
);

File: src/HookBridge.Web/src/app/core/models/doc.models.ts
export interface DocSectionDto {
  id: string;
  title: string;
  category: string;
  summary: string;
  contentMarkdown: string;
  codeSnippets: Record<string, string>;
}

export interface DocParamDto {
  name: string;
  type: string;
  required: boolean;
  description: string;
  example?: string;
}

export interface DocHeaderDto {
  name: string;
  required: boolean;
  description: string;
  example: string;
}

export interface DocEndpointDto {
  id: string;
  category: string;
  method: string;
  path: string;
  summary: string;
  description: string;
  pathParameters: DocParamDto[];
  queryParameters: DocParamDto[];
  requestHeaders: DocHeaderDto[];
  sampleRequestBody?: string;
  expectedStatusCode: number;
  sampleResponseBody?: string;
  codeSnippets: Record<string, string>;
}

export interface SdkRecipeDto {
  language: string;
  displayName: string;
  description: string;
  installationCommand: string;
  verificationSnippet: string;
  publishingSnippet: string;
  dependencies: string[];
}

export interface ApiReferenceResponse {
  version: string;
  environment: string;
  baseUrl: string;
  guides: DocSectionDto[];
  endpoints: DocEndpointDto[];
  sdkRecipes: SdkRecipeDto[];
}

export interface GenerateDocSnippetRequest {
  language: string;
  method: string;
  path: string;
  headers?: Record<string, string>;
  body?: string;
  apiKey?: string;
  signingSecret?: string;
}

export interface GenerateDocSnippetResponse {
  language: string;
  snippet: string;
  formattedCurl: string;
}

export interface SignatureVerificationResult {
  isValid: boolean;
  error?: string;
  timestamp?: number;
  clockSkewSeconds?: number;
  canonicalPayload?: string;
  computedSignature?: string;
}
